### DISCUSSION TITLE
Is there already a rule for this deprecation?

### DISCUSSION THREAD

**Discussion by parijke** [rectorphp/rector#7220] [Q&A] [answered]
The "Symfony\Component\Console\Command\Command::$defaultName" property is considered final. You should not override it in "App\Command\MyCommand".

**Comment by TomasVotruba** [comment, marked as answer]
The Symfony rules can be found here: https://github.com/rectorphp/rector-symfony/blob/main/docs/rector_rules_overview.md
Check if there is the one you're looking for :+1:

**Reply by parijke** [reply]
As far as I understand the rules, in that case the answer is not (yet)

**Reply by TomasVotruba** [reply]
I recall somebody adding some rule last month.
I've regenerated the docs as it was outdated.

What about this one?
https://github.com/rectorphp/rector-symfony/blob/main/docs/rector_rules_overview.md#commandpropertytoattributerector

**Reply by parijke** [reply]
That's the one I think. Thanks Tomas

**Reply by TomasVotruba** [reply]
You're welcome :+1:

**Reply by parijke** [reply]
This saved me 33 files by hand. Now 3 secs.
Is the book only in soft copy?

**Reply by TomasVotruba** [reply]
Haha, nice one :clap: 

Try PHP 8.0 upgrade... saved me few lives already.

Yes, we update it every ~6 months with the latest Rector tricks and configuration tips. The book should have value even for Rector 1, 2 and 3.

The udpated might be a little trick with paper ;)

**Comment by parijke** [comment]
Makes sense. Gonna order it anyways. Tjanks Tomas

Op wo 8 jun. 2022 12:44 schreef Tomas Votruba ***@***.***>:

**Reply by TomasVotruba** [reply]
Thanks! 
I wish you great reading and even better massive improvement of your codebase :+1:
